pkg/types/validation/installconfig.go

```diff
@@ -50,6 +51,24 @@ func ValidateInstallConfig(c *types.InstallConfig) field.ErrorList {
 	if c.TypeMeta.APIVersion == "" {
 		return field.ErrorList{field.Required(field.NewPath("apiVersion"), "install-config version required")}
 	}
+	if c.Proxy != nil {
+		if c.Proxy.NoProxy != "" && strings.Contains(c.Proxy.NoProxy, " ") {
+			return field.ErrorList{field.Invalid(field.NewPath("noProxy"), c.Proxy.NoProxy, fmt.Sprintf("noProxy must not have spaces"))}
```

Comment:
Don't return when an invalid field is found. Add the error to the list and continue looking for more validation errors.
```suggestion
			allErrs = append(allErrs, field.Invalid(field.NewPath("noProxy"), c.Proxy.NoProxy, fmt.Sprintf("noProxy must not have spaces")))
```